src/frontends/pytorch/src/transforms/torchfx_torchao_pattern_replacer.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "torchfx_torchao_pattern_replacer.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/split.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/util/framework_node.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "utils.hpp"
+#include "utils_quantize.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+using namespace ov::op;
+using namespace ov::pass::pattern;
+
+WeightINT4PackMMReplacer::WeightINT4PackMMReplacer() {
+    const auto& const_1 = wrap_type<v0::Constant>();
+    const auto& const_2 = wrap_type<v0::Constant>();
+    const auto& const_3 = wrap_type<v0::Constant>();
+
+    const auto& weight_int4pack_mm = wrap_type<ov::op::util::FrameworkNode>({any_input(), const_1, const_2, const_3});
```

Comment:
Why is this done as transformation and not as a simple operation converter?

Reply:
While you can do that as converter I agree that doing it as transformation makes sense